filament/backend/src/metal/MetalDriver.mm

```diff
@@ -612,8 +619,14 @@
 }
 
 SyncStatus MetalDriver::getSyncStatus(Handle<HwSync> sh) {
-    // TODO: implement Sync objects
-    return SyncStatus::SIGNALED;
+    auto* fence = handle_cast<MetalFence>(mHandleMap, sh);
+    FenceStatus status = fence->wait(0);
```

Comment:
does the correct thing happen here if we get to this point before `encode()` has been called? (i.e. it should return `NOT_SIGNALED`)

Reply:
Yes, `fence->wait(0)` will return `FenceStatus::TIMEOUT_EXPIRED` (because `state->status` has already been initialized to `TIMEOUT_EXPIRED`) and this function will then return `SyncStatus::NOT_SIGNALED`.